lib/services/local/presence.go

```diff
@@ -1700,23 +1704,45 @@ func FakePaginate(resources []types.ResourceWithLabels, req proto.ListResourcesR
 		PredicateExpression: req.PredicateExpression,
 	}
 
+	// afterMatchCount counts matches after we found the limit.
+	afterMatchCount := 0
```

Comment:
I think the logic would be simpler if this was just a `matchCount` that was zeroed when breaking early from the loop.

Reply:
i had to zero it afterwards to handle the case where if loop ends first (but i did change the logic a bit to make it cleaner)